android/engine/src/main/java/org/smartregister/fhircore/engine/util/FileUtil.kt

```diff
@@ -87,4 +89,45 @@ class FileUtil {
     val ASSET_BASE_PATH_RESOURCES =
       (System.getProperty("user.dir") + File.separator + "src" + File.separator)
   }
+
+  fun writeFileOnInternalStorage(
+    context: Context,
+    fileName: String?,
+    body: String?,
+    dirName: String
+  ) {
+
+    val dir = File(context.getFilesDir(), dirName)
+    if (!dir.exists()) {
+      dir.mkdir()
+    }
+    try {
+      val file = File(dir, fileName)
+      val writer = FileWriter(file)
+      writer.append(body)
+      writer.flush()
+      writer.close()
+    } catch (e: Exception) {
```

Comment:
can we catch a tighter exception? Don't we use Timber for reporting instead of print?

Reply:
The `FileWriter` implements a `Closeable` so you can use the Kotlin [use](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/use.html) function instead.

Reply:
@pld This addressed by implementing @ellykits suggestion